packages/workbox-precaching/controllers/PrecacheController.mjs

```diff
@@ -299,119 +201,41 @@ class PrecacheController {
     }
 
     await cacheWrapper.put({
-      cacheName: this._getTempCacheName(),
-      request: precacheEntry._cacheRequest,
-      response,
       event,
       plugins,
+      request,
+      response,
+      cacheName: this._cacheName,
     });
-
-    await this._precacheDetailsModel._addEntry(precacheEntry);
-
-    return true;
-  }
-
-  /**
-   * Compare the URLs and determines which assets are no longer required
-   * in the cache.
-   *
-   * This should be called in the service worker activate event.
-   *
-   * @return {
-   * Promise<workbox.precaching.CleanupResult>}
-   * Resolves with an object containing details of the deleted cache requests
-   * and precache revision details.
-   *
-   * @private
-   */
-  async _cleanup() {
-    const expectedCacheUrls = [];
-    this._entriesToCacheMap.forEach((entry) => {
-      const fullUrl = new URL(entry._cacheRequest.url, location).toString();
-      expectedCacheUrls.push(fullUrl);
-    });
-
-    const [deletedCacheRequests, deletedRevisionDetails] = await Promise.all([
-      this._cleanupCache(expectedCacheUrls),
-      this._cleanupDetailsModel(expectedCacheUrls),
-    ]);
-
-    if (process.env.NODE_ENV !== 'production') {
-      printCleanupDetails(deletedCacheRequests, deletedRevisionDetails);
-    }
-
-    return {
-      deletedCacheRequests,
-      deletedRevisionDetails,
-    };
   }
 
   /**
-   * Goes through all the cache entries and removes any that are
-   * outdated.
-   *
-   * @private
-   * @param {Array<string>} expectedCacheUrls Array of URLs that are
-   * expected to be cached.
-   * @return {Promise<Array<string>>} Resolves to an array of URLs
-   * of cached requests that were deleted.
+   * @return {Map<string, string>} The mapping of original URL to the
+   * corresponding cache key.
    */
-  async _cleanupCache(expectedCacheUrls) {
-    if (!await caches.has(this._cacheName)) {
-      // Cache doesn't exist, so nothing to delete
-      return [];
-    }
-
-    const cache = await caches.open(this._cacheName);
-    const cachedRequests = await cache.keys();
-    const cachedRequestsToDelete = cachedRequests.filter((cachedRequest) => {
-      return !expectedCacheUrls.includes(
-          new URL(cachedRequest.url, location).toString()
-      );
-    });
-
-    await Promise.all(
-        cachedRequestsToDelete.map((cacheUrl) => cache.delete(cacheUrl))
-    );
-
-    return cachedRequestsToDelete.map((request) => request.url);
+  getUrlsToCacheKeys() {
+    return this._urlsToCacheKeys;
   }
 
   /**
-   * Goes through all entries in indexedDB and removes any that are outdated.
-   *
-   * @private
-   * @param {Array<string>} expectedCacheUrls Array of URLs that are
-   * expected to be cached.
-   * @return {Promise<Array<string>>} Resolves to an array of URLs removed
-   * from indexedDB.
+   * @return {Array<string>} All of the URLs that have been precached.
```

Comment:
Is this method public? If so it should probably have a description for the docs; if it's not then it should start with an underscore so it can be minified.